astropy/io/fits/card.py

```diff
@@ -161,7 +161,7 @@ def __init__(self, keyword=None, value=None, comment=None, **kwargs):
         self._keyword = None
         self._value = None
         self._comment = None
-
+        self._valuestring = None
```

Comment:
Why was this moved?

Reply:
Because it must be defined before `self.value` is set. Using the new header class triggered this bug (https://github.com/astropy/astropy/commit/b135229602acb50451c9819a722ea09a5b598132) in the getter/setter loop for `.value`, and the same for `.comment`.